cognite/client/_api/iam.py

```diff
@@ -597,11 +598,30 @@ def revoke(self, id: int | Sequence[int]) -> Session | SessionList:
         Returns:
             Session | SessionList: List of revoked sessions. If the user does not have the sessionsAcl:LIST capability, then only the session IDs will be present in the response.
         """
-        identifiers = IdentifierSequence.load(ids=id, external_ids=None)
-        items = {"items": identifiers.as_dicts()}
 
-        result = SessionList._load(self._post(self._RESOURCE_PATH + "/revoke", items).json()["items"])
-        return result[0] if isinstance(id, int) else result
+        revoked_sessions = cast(
+            list,
+            self._delete_multiple(
+                identifiers=IdentifierSequence.load(ids=id, external_ids=None),
+                wrap_ids=True,
+                returns_items=True,
+                delete_endpoint="/revoke",
+            ),
+        )
+
+        parsed_sessions = []
+        for session in revoked_sessions:
+            parsed_sessions.append(
+                Session(
+                    id=session["id"],
+                    type=session.get("type"),
+                    status=session.get("status"),
+                    creation_time=session.get("creationTime"),
+                    expiration_time=session.get("expirationTime"),
+                    client_id=session.get("clientId"),
+                )
+            )
+        return SessionList(parsed_sessions) if len(parsed_sessions) > 1 else parsed_sessions[0]
```

Comment:
This should use `identifiers.is_singleton` so that a given list of one session id is returned as a `SessionList`, while one given session id is returned as `Session`